Source/Controllers/ESTabBarController.m

```diff
@@ -180,6 +182,17 @@ - (void)setSelectedIndex:(NSInteger)selectedIndex animated:(BOOL)animated {
     }
 }
 
+- (void)setBadgeValue:(NSString*)value atIndex:(NSInteger)index
+{
+    NSString* key = [NSString stringWithFormat:@"%ld", (long)index];
```

Comment:
For controllers and actions, I'm using `NSNumber`s as keys, instead of strings. Would be nice if we can keep that to preserve the consistency. If then, this method would look like:
```
- (void)setBadgeValue:(NSString *)value atIndex:(NSInteger)index {
    if (value) {
        self.badgeValues[@(index)] = value;
    } else {
        [self.badgeValues removeObjectForKey:@(index)];
    }

    [self updateInterfaceIfNeeded];
}
```